### ISSUE TITLE
Text is cut off in nodes

### ISSUE THREAD

**Issue by prototyperspective** [amelioro/ameliorate#557]
### Describe your issue

![text](https://github.com/user-attachments/assets/d6006f6f-59db-48d3-ac22-e01d2687f5b6)

### Solution you'd like

There are multiple possible solutions and these could be discussed here.
* The nodes could be a little bit larger if there is more text in them and how large could be configurable (I'd suggest that the default would be quite a bit larger than now; this may either itself not fully solve this or one sets a max-length for these texts so that the text always fits into the node with any additional details going into the Notes of the item)
* The node could expand when hovering over it
* The nodes could become bigger and the font-size be decreased up to the max-length of the node text

However, I think for exporting a map, the full text of all items definitely needs to be visible. If you want to export an image version one can't hover over nodes. Nodes with long texts would need to be edited to improve how the problem tree map looks like when shared with others / exported but having a few somewhat larger nodes is much better than just making parts of the text not readable. So just showing the full text when hovering would not be sufficient for that...e.g. an export to image view would adjust the nodes sizes and then one could also have an option whether the full text should always be shown (I would enable that).

Maybe there are more ways this could be addressed.

### Additional context

Mentioned in #535 where @keyserj wrote
> My intention was to keep node text small and concise so they can be quickly glanced at to understand high-level concepts. To this end, nodes are fixed size, and extra explanatory text can be included in the node's notes. This is open to discussion. It probably requires some practice for wording/sentence-fragmenting, but I've always been able to word nodes such that their concept fits in the current node's size.

Facilitating users keep texts in nodes short and succinct is good but there still needs to be a way to see the full text.

If text is entered longer than the max-length it should display an error and save the text up to the max length and ask the user  to shorten/summarize it and to put any additional info into the notes (Maybe that field was better be called "More information" or "More information and notes" or similar.)

**Comment by keyserj** [comment]
> Text is cut off in nodes

As I was quoted, I prefer keeping the nodes small to encourage sticking to "node = concept" not "node = sentence/explanation". But I think UX could be a little more forgiving than the current cut-off + have-to-scroll (scrolling such a small text box sucks).

> font-size be decreased

Shrinking font-size to fit the text sounds more appealing to me than increasing node size. This way, if the text barely overflows, the UX is very forgiving, but becomes more discouraging the more text overflows. This aligns with the idea that more overflow = worse for understandability. Doesn't seem like this is a super-standard strategy, but there are a few libraries for this that seem decent [here](https://stackoverflow.com/a/77154779).

> However, I think for exporting a map, the full text of all items definitely needs to be visible

This is a good point.

> If text is entered longer than the max-length it should display an error and save the text up to the max length and ask the user to shorten/summarize it and to put any additional info into the notes

Would have to design UX for this, though it seems very useful for conveying the intent. Might do this separately from the functional change.

> (Maybe that field was better be called "More information" or "More information and notes" or similar.)

Yeah "more information" might be clearer than "notes"

**Comment by prototyperspective** [comment]
> could be a little more forgiving than the current cut-off + have-to-scroll

Yes, that but also it needs a better way to see the full text than scrolling in a tiny node. This is the main problem: the way to see the full text whenever the text is longer than whatever is the max is not sleek but pretty inconvenient.

>but becomes more discouraging the more text overflows

I think if the text gets too small and probably also in general a way to just at least temporarily increase the node size is needed – this is not so much about people reading the problem maps they created themselves but people reading other people's problem maps. It's already a problem if the full text can't be seen by default so it already well signifies there's a problem and facilitates people to shorten the text.

**Comment by keyserj** [comment]
> if the text gets too small and probably also in general a way to just at least temporarily increase the node size is needed

Fair point. An easy interactive solution could be to allow hovering to show all the text in a tooltip.

I also see that increasing node size may be the only way to address this problem for a not-interactive screenshot. This just seems like a bit of an annoying thing to support. In theory, maybe you could also increase the screenshot resolution and just zoom in with whatever screenshot-viewing software you're using.

**Comment by keyserj** [comment]
Implemented, demo below. At max reduction, four lines fit, which I think helps a lot. Going beyond the max reduction and four lines no longer has a scrollbar, mainly so font size calcs don't need to remove it to check fit if it exists, but I think it's fine because at this point there's very likely more text than necessary and I don't think it's worth supporting.

I plan to add a warning popup when font reduction happens for the user's first time, to warn about performance concerns when more than tens of displayed nodes have font reduced.

https://github.com/user-attachments/assets/1b764877-dcf3-4e5c-8afd-72551990c522
